tests/queries/0_stateless/01600_select_in_different_types.sql

```diff
@@ -0,0 +1,30 @@
+-- SELECT 1 IN (SELECT -1)
+-- SELECT -1 IN (SELECT 1)
+
+DROP TABLE IF EXISTS select_in_test;
+
+CREATE TABLE select_in_test(value UInt8) ENGINE=TinyLog;
+INSERT INTO select_in_test VALUES (1), (2), (3);
+
+SELECT value FROM select_in_test WHERE value IN (-1);
+SELECT value FROM select_in_test WHERE value IN (SELECT -1);
+
+SELECT value FROM select_in_test WHERE value IN (1);
+SELECT value FROM select_in_test WHERE value IN (SELECT 1);
+
+DROP TABLE select_in_test;
+
+CREATE TABLE select_in_test(value Int8) ENGINE=TinyLog;
+INSERT INTO select_in_test VALUES (-1), (2), (3);
+
+SELECT value FROM select_in_test WHERE value IN (1);
+SELECT value FROM select_in_test WHERE value IN (SELECT 1);
+
+SELECT value FROM select_in_test WHERE value IN (2);
+SELECT value FROM select_in_test WHERE value IN (SELECT 2);
+
+DROP TABLE select_in_test;
+
+SELECT 1 IN (1);
+-- Right now this working because of accurate cast. Need to discuss.
+SELECT '1' IN (SELECT 1);
```

Comment:
It's ok to keep.

Reply:
It's more aligned with the behaviour of standard SQL where weak typing is used.
We don't like weak typing :) But we tend to make our SQL more compatible with standard (through some settings or without). If this behaviour is "for free", let's keep it without introducing any settings.

Reply:
@alexey-milovidov It is for free for cases where we insert data in set from subselect, but when we make set from collection during AST visiting, we use convertFieldToType and it will throw exception if types does not match.

Example: SELECT '1' IN (1); will throw exception.

It does not make a lot of changes to update this behaviour. Should we fix it ?

Reply:
It looks inconsistent but let's keep it as is for now.